temporal-sdk/src/main/java/io/temporal/internal/sync/ActivityExecutionContextImpl.java

```diff
@@ -228,6 +239,17 @@ public boolean isDoNotCompleteOnReturn() {
     return doNotCompleteOnReturn;
   }
 
+  public boolean isUseLocalManualCompletion() {
+    return useLocalManualCompletion;
+  }
+
+  @Override
+  public ActivityCompletionClient useLocalManualCompletion() {
+    doNotCompleteOnReturn();
+    useLocalManualCompletion = true;
+    return new ActivityCompletionClientImpl(manualCompletionClientFactory, completionHandle);
```

Comment:
Make this method and other methods thread safe. It looks like doNotComplete on return should be also made thread safe.

Reply:
Done